design/vela-core/apply-once-only.md

```diff
@@ -0,0 +1,88 @@
+# Apply-Once-Only: Apply workload/trait only when spec is changed
+
+- Owner: Yue Wang(@captainroy-hy), Jianbo Sun(@wonderflow)
+- Date: 11/24/2020
+- Status: Implemented
+
+## Intro
+When an ApplicationConfiguration is deployed, 
+vela-core will create(apply) corresponding workload/trait instances and keep them stay inline with the `spec` defined in ApplicationConfiguration through periodical reconciliation. 
+
+If `Apply-Once-Only` is enabled, after creating workload/trait instances, vela-core will skip reconciling them even if they are changed by others (e.g., trait controller, workload controller,etc). 
+Since the create operation is the only one apply operation occurring on workload/trait, we call this mechanism as `Apply-Once-Only`.
```

Comment:
```suggestion
Since the create operation is the only one apply operation occurring on workload/trait, we call this mechanism as `Apply Once Only`.
```